test/functional/webauthn_verifications_controller_test.rb

```diff
@@ -258,5 +218,84 @@ class WebauthnVerificationsControllerTest < ActionController::TestCase
         assert_equal "The token in the link you used has either expired or been used already.", flash[:alert]
       end
     end
+
+    context "when no port is given" do
+      setup do
+        @challenge = session[:webauthn_authentication]["challenge"]
```

Comment:
Hmm I see lots of examples in the codebase where we use `dig` in the controller but no examples of where we use it in the corresponding test. That kind of makes sense to me, since while the safe navigation feature of dig is great in the controller, in the test it would be more helpful if it blew up when one of the hash values wasn't present, as opposed to returning nil which would obscure the error.

Reply:
Oh, good point!